src/libs/Navigation/Navigation.js

```diff
@@ -91,6 +92,27 @@ function dismissModal(shouldOpenDrawer = false) {
     openDrawer();
 }
 
+/**
+ * Capture any navigation attempts until the navigation container and all it's children finish mounting
+ *
+ * @param {String} route
+ */
+function navigateLater(route) {
+    lastAttemptedRoute = route;
+}
+
+/**
+ * Enable navigation after the navigation tree becomes ready
+ * Executes the last captured navigation command if attempts were made prior being ready
+ */
+function enableNavigation() {
+    if (lastAttemptedRoute) {
+        navigate(lastAttemptedRoute);
+    }
+    // eslint-disable-next-line no-use-before-define
+    exports.navigate = navigate;
+}
+
```

Comment:
This didn't resolve the original issue, and perhaps would not be a useful addition

Here's what happens following the Test steps in the PR

1. Start logged out - The Navigation is immediately enabled because we have an active navigator - `PublicScreens`
2. While logged out, type and login with brand new email address so it can trigger sign up flow
3. Get validation link from e-mail. On dev I'm copying the email link and appending it to localhost - e.g. http://localhost:8080/setpassword/10024706/EHOHMPGWM
4. Type new password and click "set password" - at this point `PublicScreens` unmount, but `AuthScreens` are not yet mounted. An error happens as a call to `navigate` is made during this time

Reply:
Sounds like we'd need to hook into some other `onReady` situation like the sidebar mounting or something to determine "readiness". But if we can avoid this complexity then let's do that instead :)